### PR TITLE
Feature/app test

### LINKED ISSUE #50: Finish App Test
- [ ] all possible unit tests
- [ ] all integration tests
- [ ] sad paths

### PR DESCRIPTION
**Pull request by holladayian**
## Description
This PR adds a few unit and integration tests, complete with mocked functions into App.test.js.

## Context
I know some of them might seem a little out of order, but it should robustly test for units and integrations as well.

## Where To Start
`App.test.js`

## Affected areas of application
This literally just affects `App.test.js` as mentioned above.

## Screenshots (if applicable):
no

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] It's literally a test file that passes 

## Relevant Tickets
Closes #50 

## Questions or Notes (if applicable):
Any other things y'all would like to test for?

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -12,20 +12,20 @@ import ioc from 'socket.io-client';
...
     this.state = {
       attempts: [],
       chat: [],
       display: [],
-      inGame: false,
-      isGameReady: false,
-      isGenerator: null,
+      inGame: this.props.inGame || false,
+      isGameReady: this.props.isGameReady || false,
+      isGenerator: this.props.isGenerator || null,
```
Comment (Chriscastanuela) [inline src/App.js:23]: dynamic 🔥

File: src/App.test.js [hunk 2]
```diff
@@ -3,65 +3,255 @@ import { App } from './App.js';
...
-
-    let homepage = screen.getByTestId('homepage');
+    
+    let banger = screen.getByText('BangWords');
+    let guest = screen.getByText('Logged in as: guest!');
+    let lobby = screen.getByTestId('lobby');
+    
+    expect(banger).toBeInTheDocument();
+    expect(guest).toBeInTheDocument();
+    expect(lobby).toBeInTheDocument();
```
Comment (Chriscastanuela) [inline src/App.test.js:29]: Very cool integration test

**Review by Chriscastanuela — APPROVED** [review]
Great stuff, looks like a lot of work

**Review by GreyMatteOr — APPROVED** [review]
I love all the mocking. Finally everything you learned in kindergarten really paid off!

**Pull request merged**